controllers/usersignup/usersignup_controller_test.go

```diff
@@ -4259,6 +4259,110 @@ func TestUserSignupStatusNotReady(t *testing.T) {
 	})
 }
 
+// Test the scenario where the user trying to reactivate but their old space still exists
+func TestUserReactivatingWhileOldSpaceExists(t *testing.T) {
+	// given
+	meta := commonsignup.NewUserSignupObjectMeta("", "john.doe@redhat.com")
+	userSignup := &toolchainv1alpha1.UserSignup{
+		ObjectMeta: meta,
+		Spec: toolchainv1alpha1.UserSignupSpec{
+			Userid:   "UserID123",
+			Username: meta.Name,
+		},
+		Status: toolchainv1alpha1.UserSignupStatus{
+			CompliantUsername: "john-doe",
+		},
+	}
+	mur := murtest.NewMasterUserRecord(t, "john-doe", murtest.MetaNamespace(test.HostOperatorNs))
+	mur.Labels = map[string]string{
+		toolchainv1alpha1.MasterUserRecordOwnerLabelKey: userSignup.Name,
+		toolchainv1alpha1.UserSignupStateLabelKey:       "approved",
+	}
+
+	space := spacetest.NewSpace("john-doe",
+		spacetest.WithSpecTargetCluster("member-1"),
+		spacetest.WithStatusTargetCluster("member-1"), // already provisioned on a target cluster
+		spacetest.WithFinalizer(),
+		spacetest.WithDeletionTimestamp())
```

Comment:
don't think it matters, but probably it's more clear that space is terminating ?
 
```suggestion
		spacetest.WithCondition(spacetest.Terminating()),
		spacetest.WithDeletionTimestamp())
```